File: app/src/main/java/com/example/myapplication/ChatAdapter.java
package com.example.myapplication;

import android.graphics.Color;
import android.util.TypedValue;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

public class ChatAdapter extends RecyclerView.Adapter<ChatAdapter.ChatViewHolder> {

    private List<ChatMessage> chatList;
    private OnSuggestionClickListener listener;

    public interface OnSuggestionClickListener {
        void onSuggestionClick(String text);
    }

    public ChatAdapter(List<ChatMessage> chatList, OnSuggestionClickListener listener) {
        this.chatList = chatList;
        this.listener = listener;
    }

    @Override
    public int getItemViewType(int position) {
        return chatList.get(position).getType();
    }

    @NonNull
    @Override
    public ChatViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        int layoutRes = (viewType == ChatMessage.TYPE_MINE) ? R.layout.item_chat_mine : R.layout.item_chat_other;
        View view = LayoutInflater.from(parent.getContext()).inflate(layoutRes, parent, false);
        return new ChatViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ChatViewHolder holder, int position) {
        ChatMessage msg = chatList.get(position);

        // [핵심 수정] 텍스트 갱신 보장
        if (holder.tvContent != null) {
            String content = msg.getContent();
            if (content != null && !content.isEmpty()) {
                holder.tvContent.setText(content);
                holder.tvContent.setVisibility(View.VISIBLE);
            } else {
                holder.tvContent.setVisibility(View.GONE);
            }
        }

        // 이미지 표시 로직
        if (holder.ivImage != null) {
            if (msg.isImage() && msg.getImageBitmap() != null) {
                holder.ivImage.setVisibility(View.VISIBLE);
                holder.ivImage.setImageBitmap(msg.getImageBitmap());
            } else {
                holder.ivImage.setVisibility(View.GONE);
            }
        }

        // 추천 질문(칩) 생성 로직
        if (holder.layoutSuggestions != null) {
            holder.layoutSuggestions.removeAllViews(); // 이전 뷰들 삭제
            List<String> suggestions = msg.getSuggestions();
            if (suggestions != null && !suggestions.isEmpty()) {
                holder.layoutSuggestions.setVisibility(View.VISIBLE);
                for (String text : suggestions) {
                    TextView chip = new TextView(holder.itemView.getContext());
                    chip.setText(text);
                    chip.setTextSize(14);
                    chip.setTextColor(Color.parseColor("#1E88E5"));
                    chip.setBackgroundResource(R.drawable.bg_suggestion_chip);
                    chip.setPadding(30, 15, 30, 15); // 패딩 직접 추가로 시인성 확보
                    chip.setGravity(Gravity.CENTER);

                    LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(
                            ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
                    params.setMargins(0, 0, 15, 15);
                    chip.setLayoutParams(params);

                    chip.setOnClickListener(v -> listener.onSuggestionClick(text));
                    holder.layoutSuggestions.addView(chip);
                }
            } else {
                holder.layoutSuggestions.setVisibility(View.GONE);
            }
        }
    }

    @Override
    public int getItemCount() { return chatList.size(); }

    static class ChatViewHolder extends RecyclerView.ViewHolder {
        TextView tvContent;
        ImageView ivImage;
        LinearLayout layoutSuggestions;

        ChatViewHolder(View itemView) {
            super(itemView);
            tvContent = itemView.findViewById(R.id.tvContent);
            ivImage = itemView.findViewById(R.id.ivImage);
            layoutSuggestions = itemView.findViewById(R.id.layoutSuggestions);
        }
    }
}

File: app/src/main/java/com/example/myapplication/guardian/onboarding/ChatAdapter.kt
package com.example.myapplication.guardian.onboarding

import android.graphics.Color
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.myapplication.R
import com.example.myapplication.common.model.ChatMessage

class ChatAdapter(
    private val chatList: List<ChatMessage>,
    private val listener: OnSuggestionClickListener
) : RecyclerView.Adapter<ChatAdapter.ChatViewHolder>() {

    // 인터페이스 정의 (기존 이름 유지)
    interface OnSuggestionClickListener {
        fun onSuggestionClick(text: String?)
    }

    override fun getItemViewType(position: Int): Int {
        return chatList[position].type
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChatViewHolder {
        val layoutRes = if (viewType == ChatMessage.TYPE_MINE) {
            R.layout.item_chat_mine
        } else {
            R.layout.item_chat_other
        }
        val view = LayoutInflater.from(parent.context).inflate(layoutRes, parent, false)
        return ChatViewHolder(view)
    }

    override fun onBindViewHolder(holder: ChatViewHolder, position: Int) {
        val msg = chatList[position]

        // 1. 텍스트 갱신 보장
        holder.tvContent?.let { tv ->
            val content = msg.content
            if (!content.isNullOrEmpty()) {
                tv.text = content
                tv.visibility = View.VISIBLE
            } else {
                tv.visibility = View.GONE
            }
        }

        // 2. 이미지 표시 로직
        holder.ivImage?.let { iv ->
            if (msg.isImage && msg.imageBitmap != null) {
                iv.visibility = View.VISIBLE
                iv.setImageBitmap(msg.imageBitmap)
            } else {
                iv.visibility = View.GONE
            }
        }

        // 3. 추천 질문(칩) 생성 로직
        holder.layoutSuggestions?.let { layout ->
            layout.removeAllViews() // 이전 뷰들 삭제
            val suggestions = msg.suggestions
            if (!suggestions.isNullOrEmpty()) {
                layout.visibility = View.VISIBLE
                for (text in suggestions) {
                    val chip = TextView(holder.itemView.context).apply {
                        this.text = text
                        textSize = 14f
                        setTextColor(Color.parseColor("#1E88E5"))
                        setBackgroundResource(R.drawable.bg_suggestion_chip)
                        setPadding(30, 15, 30, 15)
                        gravity = Gravity.CENTER

                        val params = LinearLayout.LayoutParams(
                            ViewGroup.LayoutParams.WRAP_CONTENT,
                            ViewGroup.LayoutParams.WRAP_CONTENT
                        ).apply {
                            setMargins(0, 0, 15, 15)
                        }
                        layoutParams = params

                        setOnClickListener { listener.onSuggestionClick(text) }
                    }
                    layout.addView(chip)
                }
            } else {
                layout.visibility = View.GONE
            }
        }
    }

    override fun getItemCount(): Int = chatList.size

    // ViewHolder 클래스 (기존 필드명 유지)
    class ChatViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvContent: TextView? = itemView.findViewById(R.id.tvContent)
        val ivImage: ImageView? = itemView.findViewById(R.id.ivImage)
        val layoutSuggestions: LinearLayout? = itemView.findViewById(R.id.layoutSuggestions)
    }
}